### PR TITLE
Add retries to GetHTTPResponse()

### PR DESCRIPTION
**Pull request by mikeharder**
- Addresses #245

Recommend ignoring whitespace: https://github.com/aspnet/KoreBuild/pull/246/files?w=1

### CODE REVIEW

**Comment by mikeharder** [comment]
@pranavkm: I see that KoreBuild uses Travis to verify on Mac/Linux, but not AppVeyor to verify on Windows.  Should it?

File: build/dotnet/dotnet-install.ps1 [hunk 1]
```diff
@@ -131,42 +150,45 @@ function Load-Assembly([string] $Assembly) {
...
+    {
+        $HttpClient = $null
+
+        try {
+            # HttpClient is used vs Invoke-WebRequest in order to support Nano Server which doesn't support the Invoke-WebRequest cmdlet.
+            Load-Assembly -Assembly System.Net.Http
+            if($ProxyAddress){
+                $HttpClientHandler = New-Object System.Net.Http.HttpClientHandler
+                $HttpClientHandler.Proxy =  New-Object System.Net.WebProxy -Property @{Address=$ProxyAddress}
+                $HttpClient = New-Object System.Net.Http.HttpClient -ArgumentList $HttpClientHandler
```
Comment (pranavkm) [inline build/dotnet/dotnet-install.ps1:163]: Should we be re-using the `HttpClient`? Presumably guidance on re-using `HttpClient` instances applies here too?

**Review by pranavkm — APPROVED** [review]

**Comment by pranavkm** [comment]
Pinging @natemcmaster added the Travis verification.

**Comment by natemcmaster** [comment]
AppVeyor should have run. See https://ci.appveyor.com/project/aspnetci/korebuild. I'm going to close/reopen to see if that triggers appveyor.

**Comment by natemcmaster** [comment]
That did the trick. https://ci.appveyor.com/project/aspnetci/korebuild/build/1.0.102

File: build/dotnet/dotnet-install.ps1 (hunk 1 above)
Comment (mikeharder) [reply build/dotnet/dotnet-install.ps1:163]: I believe the main reason to re-use HttpClient is for perf when doing many requests, which shouldn't matter in this scenario.

**Comment by mikeharder** [comment]
Thanks for adding AppVeyor, I will merge once it passes.

**Pull request merged**